File: Competitive_Programming/Recursion/StringToInt.java
package Competitive_Programming.Recursion;
import java.util.Scanner;

public class StringToInt {
    static Scanner sc = new Scanner(System.in);
    static int strToInt(String n) {
        if(n.length() == 1) {
            return n.charAt(0) - '0';
        }
        int smallProb = strToInt(n.substring(0, n.length()-1));
        int lastDigit = n.charAt(n.length()-1) - '0';
        return smallProb * 10 + lastDigit;
    }
    public static void main(String[] args) {
        System.out.print("Enter String: ");
        String n = sc.next();
        System.out.print("The Integer: " + strToInt(n));
    }
}

File: Competitive_Programming/Recursion/StringToInt.py
# Recursive function to Convert String to Integer

def strToInt(n):
    if(len(n) == 1):
        return ord(n[0]) - ord('0')
    
    smallProb = strToInt(n[1:])
    firstDigit = ord(n[0]) - ord('0')
    return firstDigit * (10**(len(n)-1)) + smallProb
    
# Main
n = input("Enter String: ")
print("The Integer:" , strToInt(n))